### PR TITLE
Update Python CI to Python 3.9

### PR DESCRIPTION
**Pull request by dgelessus**
As well as some small improvements/simplifications to the prepare-python script.

I assume this will also need an update in https://github.com/kaitai-io/kaitai_ci_ui so that the results for Python 3.9 are displayed correctly?

### CODE REVIEW

**Comment by generalmimon** [comment]
> I assume this will also need an update in https://github.com/kaitai-io/kaitai_ci_ui so that the results for Python 3.9 are displayed correctly?

Yes.

File: prepare-python [hunk 1]
```diff
@@ -4,7 +4,4 @@
 
 git clone https://github.com/kaitai-io/kaitai_struct_python_runtime "$PYTHON_RUNTIME_DIR"
 
-pip install enum34
-
-# this package dropped support for Python 2.7 since 3.0.0, so we're using the latest compatible version
-pip install "unittest-xml-reporting~=2.5.2"
+python -m pip install enum34 unittest-xml-reporting
```
Comment (generalmimon) [inline prepare-python:7]: So we're dropping support for Python 2.7 here? Please state it in the PR message, if it was the intent.

However, i'd prefer to do it in another PR, because we haven't reached a consensus on https://github.com/kaitai-io/kaitai_struct/issues/821 yet.

**Review by generalmimon — CHANGES_REQUESTED** [review]

File: prepare-python (hunk 1 above)
Comment (dgelessus) [reply prepare-python:7]: No - `python -m pip install unittest-xml-reporting` now works correctly on Python 2.7 as well (I tested). Quoting my commit message from 790df3c440c08d5c6cfb5cd10a3485f38ee8195b:

```
Remove no longer needed version constraint for unittest-xml-reporting

The package now has correct Python version compatibility metadata,
so pip will automatically install the latest version compatible with
the current Python version (i. e. 2.5.2 on Python 2.7 and 3.4, and the
latest available version everywhere else) without a manual version
constraint.
```

Comment (generalmimon) [reply prepare-python:7]: When you're updating the Construct target as well, would you also simplify `prepare-construct` the same way?

https://github.com/kaitai-io/ci_targets/blob/9a4fe530890ebe1f4c112a7cbe9e89bb5bd0f64b/prepare-construct#L7-L8

Comment (dgelessus) [reply prepare-python:7]: Ah, I didn't see that Construct has a separate prepare script - will update that as well.

**Review by generalmimon — APPROVED** [review]
Thanks, merging in!
